enforcer_cached.go

```diff
@@ -93,3 +93,36 @@ func (e *CachedEnforcer) setCachedResult(key string, res bool) {
 func (e *CachedEnforcer) InvalidateCache() {
 	e.m = make(map[string]bool)
 }
+
+/* Clear cache after update Policy*/
+// AddPolicy adds an authorization rule to the current policy.
+// If the rule already exists, the function returns false and the rule will not be added.
+// Otherwise the function returns true by adding the new rule.
+func (e *CachedEnforcer) AddPolicy(params ...interface{}) (bool, error) {
+	ok, err := e.AddNamedPolicy("p", params...)
+	e.InvalidateCache()
+	return ok, err
+}
+
+// AddPolicies adds authorization rules to the current policy.
+// If the rule already exists, the function returns false for the corresponding rule and the rule will not be added.
+// Otherwise the function returns true for the corresponding rule by adding the new rule.
+func (e *CachedEnforcer) AddPolicies(rules [][]string) (bool, error) {
+	ok, err := e.AddNamedPolicies("p", rules)
+	e.InvalidateCache()
+	return ok, err
+}
+
+// RemovePolicy removes an authorization rule from the current policy.
+func (e *CachedEnforcer) RemovePolicy(params ...interface{}) (bool, error) {
+	ok, err := e.RemoveNamedPolicy("p", params...)
+	e.InvalidateCache()
+	return ok, err
+}
+
+// RemovePolicies removes authorization rules from the current policy.
+func (e *CachedEnforcer) RemovePolicies(rules [][]string) (bool, error) {
```

Comment:
How about `RemoveFilteredPolicy`?

Reply:
I don't quite understand. Does it mean to add this function?

Reply:
you need to clear cache in RmoveFilteredPolicy